### PR TITLE
New detected host mode container property on enforcer

### PR DESCRIPTION
**Pull request by amitlimaye**
Added a new property on the enforcer object to be set when we detect host mode 
containers that the enforcer is not going to enforce (ignore)

----

#

### CODE REVIEW

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "2d42282568c76def9a57efc3da363afb92968d00",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "0bbe04c3a7694b2163221cb2d7ccb5e6cfa4c57b",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "b40c998fcb0b52e2efc337a4998f7f66776d2122",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "0bbe04c3a7694b2163221cb2d7ccb5e6cfa4c57b",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "b40c998fcb0b52e2efc337a4998f7f66776d2122",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "0bbe04c3a7694b2163221cb2d7ccb5e6cfa4c57b",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "b40c998fcb0b52e2efc337a4998f7f66776d2122",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "0bbe04c3a7694b2163221cb2d7ccb5e6cfa4c57b",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "b40c998fcb0b52e2efc337a4998f7f66776d2122",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "0bbe04c3a7694b2163221cb2d7ccb5e6cfa4c57b",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "b40c998fcb0b52e2efc337a4998f7f66776d2122",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "51a05f65ef22850d6db462f948d1e21aac964385",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "7c738ab5a4038163d65509e2592b98b041b79f43",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "51a05f65ef22850d6db462f948d1e21aac964385",
    "pipeline": "master"
  }
]
```

File: specs/enforcer.spec [hunk 1]
```diff
@@ -298,6 +298,15 @@ attributes:
     read_only: true
     autogenerated: true
     transient: true
+  - name: detectedHostModeContainer
+    description: |-
+      This field indicate whether the enforcer has detected host mode containers.
```
Comment (primalmotion) [inline specs/enforcer.spec:303]: Indicates

File: specs/enforcer.spec [hunk 2]
```diff
@@ -298,6 +298,15 @@ attributes:
...
+    type: boolean
+    exposed: true
+    stored: true
+    read_only: false
+    autogenerated: true
```
Comment (primalmotion) [inline specs/enforcer.spec:308]: This cannot be autogenerated or transient

File: specs/enforcer.spec [hunk 3]
```diff
@@ -298,6 +298,15 @@ attributes:
     read_only: true
     autogenerated: true
     transient: true
+  - name: detectedHostModeContainer
```
Comment (primalmotion) [inline specs/enforcer.spec:301]: should be plural

**Review by primalmotion — CHANGES_REQUESTED** [review]

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "7c738ab5a4038163d65509e2592b98b041b79f43",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "2df665229d1acb7bc802ae970818c4d45eee1c16",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "ec56641c9c0097808a2df989b43141839986a650",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "2df665229d1acb7bc802ae970818c4d45eee1c16",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "99fb5dd35eb90e65e7ca117d3d552dc203596f36",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "2df665229d1acb7bc802ae970818c4d45eee1c16",
    "pipeline": "master"
  }
]
```

File: specs/enforcer.spec [hunk 4]
```diff
@@ -347,8 +349,7 @@ relations:
         example_value: 1000
 
       - name: enforcementStatus
-        description: If set, changes the enforcement status of the enforcer along
-          with the poke.
+        description: If set, changes the enforcement status of the enforcer along with the poke.
```
Comment (primalmotion) [inline specs/enforcer.spec:352]: There's a lot of unecessary changes here. you should need to use the `rego format` command if you are not using the vscode extension

**Review by primalmotion — CHANGES_REQUESTED** [review]
looks ok, but it's not correctly formatted

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "99fb5dd35eb90e65e7ca117d3d552dc203596f36",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "b577903d807026f444e56ed31e2b6a1c1031e672",
    "pipeline": "master"
  }
]
```

**Review by Rohit4793 — APPROVED** [review]

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "1d6180aa85a3c7a50227c875a39fc5cd53eb03ae",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "b577903d807026f444e56ed31e2b6a1c1031e672",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "1d6180aa85a3c7a50227c875a39fc5cd53eb03ae",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "b577903d807026f444e56ed31e2b6a1c1031e672",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "44f248d7c3159a00574320da203582a94f0bef37",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "b577903d807026f444e56ed31e2b6a1c1031e672",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "3ff9080d0f8566110c406d02463f24a603acc689",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "b577903d807026f444e56ed31e2b6a1c1031e672",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "3e17d26675a9aeee8aff20806329b92b099e8eda",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "b577903d807026f444e56ed31e2b6a1c1031e672",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "3e17d26675a9aeee8aff20806329b92b099e8eda",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "b32f0ae12db3377037eec8e3209b1d39b46a5f06",
    "pipeline": "master"
  }
]
```

**Review by t00f — APPROVED** [review]
LGTM

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2311",
    "commit-sha": "3e17d26675a9aeee8aff20806329b92b099e8eda",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "623",
    "commit-sha": "af0cfa0751fc14e32bf8c74d2dfa7d5f1eee94d1",
    "pipeline": "master"
  }
]
```

**Review by ericrpowers — DISMISSED** [review]

**Review by primalmotion — DISMISSED** [review]

**Review by primalmotion — APPROVED** [review]

**Pull request merged**